### PR TITLE
plan b to solve tag conflict 

### PR DESCRIPTION
**Pull request by guoshencheng**
Use `<blessed-xxx / >` instead of `<xxx />`
And add a typing definition [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30340) at [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped)

To be honest. the plan is not so excellence. but It's my final plan.

### CODE REVIEW

**Comment by guoshencheng** [comment]
@Yomguithereal my  [PR](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/30340) at [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) has been merged.

**Comment by yuchi** [comment]
So, you made DefinitelyTyped accept a PR for a definition that covers this PR, while it has not yet accepted?

**Comment by guoshencheng** [comment]
@yuchi the PR at DefinitelyTyped has been merged. i don not know why the DefinitelyTyped merged without author 's agreement

**Comment by yuchi** [comment]
Well, now TypeScript users will think that they have to use `blessed-**` in their JSX, while this is not supported at all. To avoid confusion from users I believe @Yomguithereal must now quickly react to this issue, find a solution, and (eventually) provide a patch to DefinitelyTyped.

Not a big problem IMO, but you are forcing the hand on @Yomguithereal a bit :)

**Comment by yuchi** [comment]
Also, I don‘t believe that `plan b` is a good commit message 😅

**Comment by Yomguithereal** [comment]
What's more I am now completely lost :). Can someone recap all this?

**Comment by yuchi** [comment]
My opinion in short: after a good commit squash (sorry I’m a git history purist!) this PR solves the issue perfectly at the cost of some more typing for users.

The only other way would be to go in the [react-pdf](https://github.com/diegomura/react-pdf) way:

```js
// react-blessed
export const Box = 'box';
export const Button = 'button';
// ...
```

```js
// some-app.js
import React from 'react';
import { render, Box, Button } from 'react-blessed';
render(<Box … />)
```

**Comment by Yomguithereal** [comment]
I don't like much the

```js
import { Box, Button } from 'react-blessed';
```

solution. It's hard to maintain and tedious. But typing `blessed-button` also is in a way. The way typescript interacts with react and/or jsx seems very precarious to me though. Anyway I would vouch for the solution of this PR. Let me just review the PR. (On a side note, there is no scheme such as the jsx pragma for TypeScript to handle those kind of cases gracefully?)

**Comment by Yomguithereal** [comment]
So is the PR ok for everyone involved? Can I merge and release in a near future?

File: src/fiber/fiber.js [hunk 1]
```diff
@@ -37,6 +37,10 @@ const createBlessedRenderer = function(blessed) {
       internalInstanceHandle : Object
     ) {
       const {children, ...appliedProps} = solveClass(props);
+      const blessedPrefixRegexp = /^blessed-/;
+      if (blessedPrefixRegexp.test(type)) {
```
[the remark this answers is not in the document]
Comment (guoshencheng) [reply src/fiber/fiber.js]: I have update the code as your suggestion.

**Comment by guoshencheng** [comment]
> So is the PR ok for everyone involved? Can I merge and release in a near future?

It's ok

**Comment by yuchi** [comment]
Perfect :+1:

**Comment by yuchi** [comment]
But I suggest @Yomguithereal to squash all commits into a single one, or use an explicit merge commit.

**Comment by Yomguithereal** [comment]
v0.4.0 is now released on npm.

**Comment by bensleveritt** [comment]
For what it's worth, this is in part `@types/react`'s fault. The types in there are too DOM specific (considering React is used without the DOM in many places). [There is an effort to fix this](https://blog.pshrmn.com/entry/trouble-with-react-types/), which when done will allow `react-blessed` types to reuse DOM element names.

**Pull request merged**